qiskit_experiments/framework/base_experiment.py

```diff
@@ -479,9 +476,27 @@ def set_analysis_options(self, **fields):
         )
         self.analysis.options.update_options(**fields)
 
-    def _postprocess_transpiled_circuits(self, circuits: List[QuantumCircuit], **run_options):
-        """Additional post-processing of transpiled circuits before running on backend"""
-        pass
+    def _transpile(self, circuits: List[QuantumCircuit], backend: Backend) -> List[QuantumCircuit]:
+        """Transpile the experiment circuits.
+
+        This function is used to transpile the experiment circuits before sending them to
+        the backend. It can be overridden to define custom transpilation.
+        """
+        transpile_opts = copy.copy(self.transpile_options.__dict__)
+        transpile_opts["initial_layout"] = list(self.physical_qubits)
+        transpiled = transpile(circuits, backend, **transpile_opts)
+
+        # TODO remove this deprecation after 0.3.0 release
```

Comment:
Do we have to go through the deprecation process for private methods? (`_postprocess_transpiled_circuits` starts with an underscore).